src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -1146,6 +1146,102 @@ class TcgStatics {
+   *   + info: If set, it'll replace the end of the failed assert warning with a custom text, instead of the default "follow the stated condition(s)".
+   *   + repText: If true, params.info will override the entirety of the failed assert warning.
+   *
+   * @param filter Additional condition the filtered benched Pokémon must follow. Defaults to true (so any Pokémon).
+   *
+   */
+  static void assertAnyPokemonInPlay(params=[:], Closure filter = null) {
+    def failMessage
+    def checkedPlayer = params.opp ? opp : my
+    def checkedArea = params.benched ? checkedPlayer.benched : checkedPlayer.all
+
+    def hasPokeCnt = 0
+    [params.hasPokemonEX, params.hasPokemonGX, params.hasPokemonV, params.hasPokemonVMAX].each{hasPokeCnt += 1}
+
+    def areaFilter = {
+      if (filter == null) { true } else {
+        (
+          if (params.hasType) {it.types.contains(params.hasType)} else {true}
+        ) && (
+          if (!hasPokeCnt) {true} else {
+            (params.hasPokemonEX && it.pokemonEX) ||
+            (params.hasPokemonGX && it.pokemonGX) ||
+            (params.hasPokemonV && it.pokemonV) ||
+            (params.hasPokemonVMAX && it.pokemonVMAX)
+          }
+        ) && (
+          filter.call(it)
+        )
+      }
+    }
+
+    if (params.info + !params.repText) {
+      failMessage = params.info
+    } else {
+      def benchedString = (param.benched ? "Benched " : "")
+      def typeString = (param.hasType ? "${params.hasType} " : "")
+
+      def pokeString = benchedString + typeString
+
+      if (hasPokeCnt) {
+        i = 1
+        [
+          (params.hasPokemonEX, "Pokémon-EX"),
+          (params.hasPokemonGX, "Pokémon-GX"),
+          (params.hasPokemonV, "Pokémon V"),
+          (params.hasPokemonVMAX, "Pokémon VMAX")
+        ].each{
+          if (it[0]) {
+            pokeString += it[1] + (if (i == isPokeCnt) "" else if (i == isPokeCnt-1) " or " else ", ")
+            i += 1
+          }
+        }
+      } else {
+        pokeString += "Pokémon"
+      }
+
+      failMessage = "${params.opp ? "Your opponent doesn't" : "You don't"} have any $pokeString that ${params.info ? params.info : "follow the stated condition(s)"}"
+    }
+
+    assert checkedArea.any(areaFilter) : failMessage
+  }
+
+  static void assertMyPokemon(params=[:], Closure filter = null) {
+    params.benched = false
+    params.opp = false
+    assertAnyPokemonInPlay(params, filter)
+  }
+  static void assertOppPokemon(params=[:], Closure filter = null) {
+    params.benched = false
+    params.opp = true
+    assertAnyPokemonInPlay(params, filter)
+  }
+
+  static void assertMyBenched(params=[:], Closure filter = null) {
+    params.benched = true
+    params.opp = false
+    assertAnyPokemonInPlay(params, filter)
+  }
+  static void assertOppBenched(params=[:], Closure filter = null) {
+    params.benched = true
+    params.opp = true
+    assertAnyPokemonInPlay(params, filter)
+  }
```

Comment:
Better rewritten as

Reply:
Was there a code suggestion here? Didn't get submitted

Reply:
```suggestion
  static void assertMyAll(params=[:], Closure filter = null) {
    params.benched = false
    params.opp = false
    assertAnyPokemonInPlay(params, filter)
  }
  static void assertOppAll(params=[:], Closure filter = null) {
    params.benched = false
    params.opp = true
    assertAnyPokemonInPlay(params, filter)
  }

  static void assertMyBenched(params=[:], Closure filter = null) {
    params.benched = true
    params.opp = false
    assertAnyPokemonInPlay(params, filter)
  }
  static void assertOppBenched(params=[:], Closure filter = null) {
    params.benched = true
    params.opp = true
    assertAnyPokemonInPlay(params, filter)
  }
```

Reply:
Hmm, don't really like how it sounds but it's probably better than My/Opp Pokemon (which is a bit too vague, and PokemonInPlay for these would be a bit too much)